zoning/__init__.py

```diff
@@ -0,0 +1,24 @@
+"""
+Initialization for the OpenAI configuration module.
+
+This module ensures the OpenAI key is loaded at the start of the application.
+"""
+
+import os
+import openai
+from dotenv import load_dotenv
+
+# Get the absolute path to the .env file
+dotenv_path = os.path.join(os.path.abspath(os.path.dirname(__file__)), ".env")
+
+# Load environment variables from .env
+load_dotenv(dotenv_path)
```

Comment:
Minor note that you don't need to do this. `load_dotenv` will [look up the directory tree from the CWD until it finds a `.env` file](https://github.com/theskumar/python-dotenv/blob/main/src/dotenv/main.py#L337-L338) automatically and by default if you don't provide a path yourself. That would probably be more robust.

Reply:
Getpath with utils.py

def get_project_root() -> Path:
    repo = Repo(".", search_parent_directories=True)
    path = repo.working_tree_dir

    assert path is not None, "You are not currently in a Git repo!"

    return Path(path)